tasm/tiles/include/LayoutDatabase.h

```diff
@@ -0,0 +1,62 @@
+#ifndef TASM_LAYOUTDATABASE_H
+#define TASM_LAYOUTDATABASE_H
+
+#include <experimental/filesystem>
+#include "sqlite3.h"
+#include "TileLayout.h"
+#include "Video.h"
+#include "Files.h"
+#include "EnvironmentConfiguration.h"
+#include <gtest/gtest.h>
+
+namespace tasm {
+
+class LayoutDbConfiguration {
+public:
+    static const std::experimental::filesystem::path &LayoutDbPath() {
+        return EnvironmentConfiguration::instance().defaultLayoutDatabasePath();
+    }
+};
+
+class LayoutDatabase {
+public:
+    static std::shared_ptr<LayoutDatabase> &instance() {
+        if (instance_ == nullptr) {
+            instance_ = std::make_shared<LayoutDatabase>(LayoutDatabase());
```

Comment:
I'm not positive, but I think that the compiler may be able to figure out how to construct the LayoutDatabase based on the template, so you may be able to shorten this to `std::make_shared<LayoutDatabase>();`